### ISSUE TITLE
scrolling info text

### ISSUE THREAD

**Issue by hymie0** [atsb/RL_M#72]
Since you said you're re-writing the IO routines anyway...

One of the things that always bothered me was the "rotating" text window at the bottom of the screen.  

1. Please consider making this a scrolling window as well, so that the most recent text is always at the bottom.
2. Even better, if possible, make it a scrolling window with some history, that I can scroll back through to see messages that I wasn't paying attention to.

**Comment by atsb** [comment]
Been reading the curses API stuff for this.  Looks like curses only knows about the text on the screen and the moment it leaves the screen, it is gone.  Curses doesn't keep it in memory for later use, it's just gone.  So this won't be possible.

What I can do is create a logfile which logs all the messages so we can open the file after a save and see what's been happening.

**Comment by atsb** [comment]
So I've done basically what I said.  Though I totally removed lprcat and created a new function 'fl_display_message' which reads the string buffer, converts the char to a C++ std::string and appends it to a file (data/messages.log).  I'm also priting the Cave Level above every message so the player knows what level it happened on.

I created a new function so as to keep the messages consistent.  Only the text used by the fl_display_message function is printed whereas lprintf prints the rest (but doesn't write it).   So it's a pretty nice and clean log.

However due to the previous color changes that I removed, some strings print on multiple lines, I'll fix that later.  I will commit the change tonight and build binaries to test it.
